operator/src/test/java/oracle/kubernetes/operator/steps/ManagedServerUpIteratorStepTest.java

```diff
@@ -0,0 +1,255 @@
+  private List<Memento> mementos = new ArrayList<>();
+  private DomainPresenceInfo domainPresenceInfo = createDomainPresenceInfo();
+  private TestUtils.ConsoleHandlerMemento consoleHandlerMemento;
+
+  private DomainPresenceInfo createDomainPresenceInfo() {
+    return new DomainPresenceInfo(domain);
+  }
+
+  private Domain createDomain() {
+    return new Domain().withMetadata(createMetaData()).withSpec(createDomainSpec());
+  }
+
+  private V1ObjectMeta createMetaData() {
+    return new V1ObjectMeta().namespace(NS);
+  }
+
+  private DomainSpec createDomainSpec() {
+    return new DomainSpec().withDomainUid(UID).withReplicas(1);
+  }
+
+  /**
+   * Setup env for tests.
+   * @throws NoSuchFieldException if TestStepFactory fails to install
+   */
+  @Before
+  public void setUp() throws NoSuchFieldException {
+    mementos.add(consoleHandlerMemento = TestUtils.silenceOperatorLogger());
+    mementos.add(TestStepFactory.install());
+    testSupport.addDomainPresenceInfo(domainPresenceInfo);
+  }
+
+  /**
+   * Cleanup env after tests.
+   * @throws Exception if test support failed
+   */
+  @After
+  public void tearDown() throws Exception {
+    for (Memento memento : mementos) {
+      memento.revert();
+    }
+
+    testSupport.throwOnCompletionFailure();
+  }
+
+  @Test
+  public void verifyThat_withConcurrencyOf1_bothClusteredServersShouldStartSequentially() {
+    configureCluster(CLUSTER).withMaxConcurrentStartup(1);
+    addWlsCluster(CLUSTER, "ms1", "ms2");
+
+    invokeStepWithServerStartupInfos(createServerStartupInfosForCluster(CLUSTER,"ms1", "ms2"));
+
+    assertThat(getServers(), hasItem(Arrays.asList("ms1", "ms2")));
+    assertThat(getServers().size(), equalTo(1));
+  }
+
+  @Test
+  public void verifyThat_withConcurrencyOf0_bothClusteredServersShouldStartConcurrently() {
+    configureCluster(CLUSTER).withMaxConcurrentStartup(0);
+    addWlsCluster(CLUSTER, "ms1", "ms2");
+
+    invokeStepWithServerStartupInfos(createServerStartupInfosForCluster(CLUSTER,"ms1", "ms2"));
+
+    assertThat(getServers(), allOf(hasItem("ms1"), hasItem("ms2")));
+  }
+
+  @Test
+  public void verifyThat_withConcurrencyOf2_bothClusteredServersShouldStartConcurrently() {
+    configureCluster(CLUSTER).withMaxConcurrentStartup(2);
+    addWlsCluster(CLUSTER, "ms1", "ms2");
+
+    invokeStepWithServerStartupInfos(createServerStartupInfosForCluster(CLUSTER, "ms1", "ms2"));
+
+    assertThat(getServers(), allOf(hasItem("ms1"), hasItem("ms2")));
+  }
+
+  @Test
+  public void verifyThat_withConcurrencyOf2_4clusteredServersShouldStartIn2Threads() {
+    configureCluster(CLUSTER).withMaxConcurrentStartup(2);
+    addWlsCluster(CLUSTER, "ms1", "ms2", "ms3", "ms4");
+
+    invokeStepWithServerStartupInfos(createServerStartupInfosForCluster(CLUSTER, "ms1", "ms2", "ms3", "ms4"));
+
+    assertThat(getServers(), hasItem(Arrays.asList("ms1", "ms2", "ms3", "ms4")));
+    assertThat(getServers().size(), equalTo(2));
+  }
```

Comment:
General questions (not directly related to this unit test):

Will the managed server startup ordering fix from https://github.com/oracle/weblogic-kubernetes-operator/pull/1727 continue to apply (start 'lowest' servers first)?  I assume so?  

If the answer is yes, do these unit tests verify this?

Reply:
yes, fix from [#1727](https://github.com/oracle/weblogic-kubernetes-operator/pull/1727) should continue to apply.

These unit tests won't verify it. #1727 updates `WlsClusterConfig.getServerConfigs()` to return a sorted list of server configs, which helps to create a sorted list of `ServerStartupInfos`.  But `ManagedServerUpIteratorStepTest` here create its own  `ServerStartupInfos` arrays in the tests.